File: languages/java/tags/4.01/src/main/edu/nps/moves/dis/AngularVelocityVector.java
package edu.nps.moves.dis;

import java.util.*;
import java.io.*;
import edu.nps.moves.disenum.*;
import edu.nps.moves.disutil.*;

// Jaxb and Hibernate annotations generally won't work on mobile devices. XML serialization uses jaxb, and
// javax.persistence uses the JPA JSR, aka hibernate. See the Hibernate site for details.
// To generate Java code without these, and without the annotations scattered through the
// see the XMLPG java code generator, and set the boolean useHibernateAnnotations and useJaxbAnnotions 
// to false, and then regenerate the code

import javax.xml.bind.*;            // Used for JAXB XML serialization
import javax.xml.bind.annotation.*; // Used for XML serialization annotations (the @ stuff)
import javax.persistence.*;         // Used for JPA/Hibernate SQL persistence

/**
 * 5.2.2: angular velocity measured in radians per second out each of the entity's own coordinate axes.
 *
 * Copyright (c) 2008-2010, MOVES Institute, Naval Postgraduate School. All rights reserved.
 * This work is licensed under the BSD open source license, available at https://www.movesinstitute.org/licenses/bsd.html
 *
 * @author DMcG
 */
@Entity  // Hibernate
@Inheritance(strategy=InheritanceType.JOINED)  // Hibernate
public class AngularVelocityVector extends Object implements Serializable
{
   /** Primary key for hibernate, not part of the DIS standard */
   private long pk_AngularVelocityVector;

   /** velocity about the x axis */
   protected float  x = 0;

   /** velocity about the y axis */
   protected float  y = 0;

   /** velocity about the zaxis */
   protected float  z = 0;


/** Constructor */
 public AngularVelocityVector()
 {
 }

@Transient  // Marked as transient to prevent hibernate from thinking this is a persistent property
public int getMarshalledSize()
{
   int marshalSize = 0; 

   marshalSize = marshalSize + 4;  // x
   marshalSize = marshalSize + 4;  // y
   marshalSize = marshalSize + 4;  // z

   return marshalSize;
}


/** Primary key for hibernate, not part of the DIS standard */
@Id
@GeneratedValue(strategy=GenerationType.AUTO)
public long getPk_AngularVelocityVector()
{
   return pk_AngularVelocityVector;
}

/** Hibernate primary key, not part of the DIS standard */
public void setPk_AngularVelocityVector(long pKeyName)
{
   this.pk_AngularVelocityVector = pKeyName;
}

public void setX(float pX)
{ x = pX;
}

@XmlAttribute // Jaxb
@Basic       // Hibernate
public float getX()
{ return x; 
}

public void setY(float pY)
{ y = pY;
}

@XmlAttribute // Jaxb
@Basic       // Hibernate
public float getY()
{ return y; 
}

public void setZ(float pZ)
{ z = pZ;
}

@XmlAttribute // Jaxb
@Basic       // Hibernate
public float getZ()
{ return z; 
}


public void marshal(DataOutputStream dos)
{
    try 
    {
       dos.writeFloat( (float)x);
       dos.writeFloat( (float)y);
       dos.writeFloat( (float)z);
    } // end try 
    catch(Exception e)
    { 
      System.out.println(e);}
    } // end of marshal method

public void unmarshal(DataInputStream dis)
{
    try 
    {
       x = dis.readFloat();
       y = dis.readFloat();
       z = dis.readFloat();
    } // end try 
   catch(Exception e)
    { 
      System.out.println(e); 
    }
 } // end of unmarshal method 


/**
 * Packs a Pdu into the ByteBuffer.
 * @throws java.nio.BufferOverflowException if buff is too small
 * @throws java.nio.ReadOnlyBufferException if buff is read only
 * @see java.nio.ByteBuffer
 * @param buff The ByteBuffer at the position to begin writing
 * @since ??
 */
public void marshal(java.nio.ByteBuffer buff)
{
       buff.putFloat( (float)x);
       buff.putFloat( (float)y);
       buff.putFloat( (float)z);
    } // end of marshal method

/**
 * Unpacks a Pdu from the underlying data.
 * @throws java.nio.BufferUnderflowException if buff is too small
 * @see java.nio.ByteBuffer
 * @param buff The ByteBuffer at the position to begin reading
 * @since ??
 */
public void unmarshal(java.nio.ByteBuffer buff)
{
       x = buff.getFloat();
       y = buff.getFloat();
       z = buff.getFloat();
 } // end of unmarshal method 


 /**
  * The equals method doesn't always work--mostly it works only on classes that consist only of primitives. Be careful.
  */
 public boolean equals(AngularVelocityVector rhs)
 {
     boolean ivarsEqual = true;

    if(rhs.getClass() != this.getClass())
        return false;

     if( ! (x == rhs.x)) ivarsEqual = false;
     if( ! (y == rhs.y)) ivarsEqual = false;
     if( ! (z == rhs.z)) ivarsEqual = false;

    return ivarsEqual;
 }
} // end of class

File: old/tags/2.2/cpp/DIS/AngularVelocityVector.cpp
#include <DIS/AngularVelocityVector.h> 

using namespace DIS;


AngularVelocityVector::AngularVelocityVector():
   _x(0), 
   _y(0), 
   _z(0)
{
}

AngularVelocityVector::~AngularVelocityVector()
{
}

float AngularVelocityVector::getX() const
{
    return _x;
}

void AngularVelocityVector::setX(float pX)
{
    _x = pX;
}

float AngularVelocityVector::getY() const
{
    return _y;
}

void AngularVelocityVector::setY(float pX)
{
    _y = pX;
}

float AngularVelocityVector::getZ() const
{
    return _z;
}

void AngularVelocityVector::setZ(float pX)
{
    _z = pX;
}

void AngularVelocityVector::marshal(DataStream& dataStream) const
{
    dataStream << _x;
    dataStream << _y;
    dataStream << _z;
}

void AngularVelocityVector::unmarshal(DataStream& dataStream)
{
    dataStream >> _x;
    dataStream >> _y;
    dataStream >> _z;
}


bool AngularVelocityVector::operator ==(const AngularVelocityVector& rhs) const
 {
     bool ivarsEqual = true;

     if( ! (_x == rhs._x) ) ivarsEqual = false;
     if( ! (_y == rhs._y) ) ivarsEqual = false;
     if( ! (_z == rhs._z) ) ivarsEqual = false;

    return ivarsEqual;
 }

int AngularVelocityVector::getMarshalledSize() const
{
   int marshalSize = 0;

   marshalSize = marshalSize + 4;  // _x
   marshalSize = marshalSize + 4;  // _y
   marshalSize = marshalSize + 4;  // _z
    return marshalSize;
}

